anacron/readtab.c

```diff
@@ -134,8 +134,20 @@ register_env(const char *env_var, const char *value)
 
     var_len = (int)strlen(env_var);
     val_len = (int)strlen(value);
+    if (!var_len || !val_len) {
+        return;
+    }
+
     er = obstack_alloc(&tab_o, sizeof(env_rec));
+    if (er == NULL) {
+        die_e("Cannot allocate memory.");
+    }
+
     er->assign = obstack_alloc(&tab_o, var_len + 1 + val_len + 1);
+    if (er->assign == NULL) {
+        obstack_free(&tab_o, er);
```

Comment:
Note that die_e exits the process and does not bother deallocating other data allocated by the process. So it is really strange to deallocate this particular thing.